tests/phpunit/CRM/Contribute/Page/TabTest.php

```diff
@@ -18,22 +18,79 @@
+    parent::tearDown();
+  }
+
   /**
    * Test links render correctly for manual processor.
    *
    * @throws \API_Exception
    * @throws \CiviCRM_API3_Exception
    */
-  public function testLinks() {
-    $contactID = $this->individualCreate();
-    $recurID = ContributionRecur::create()->setValues([
+  public function testLinksManual(): void {
+    [$contactID, $recurID] = $this->setupTemplate();
+
+    $templateVariable = CRM_Core_Smarty::singleton()->get_template_vars();
+    $this->assertEquals('Mr. Anthony Anderson II', $templateVariable['displayName']);
+    $this->assertEquals("<span><a href=\"/index.php?q=civicrm/contact/view/contributionrecur&amp;reset=1&amp;id=" . $recurID . "&amp;cid=" . $contactID . "&amp;context=contribution\" class=\"action-item crm-hover-button\" title='View Recurring Payment' >View</a><a href=\"/index.php?q=civicrm/contribute/updaterecur&amp;reset=1&amp;action=update&amp;crid=1&amp;cid=3&amp;context=contribution\" class=\"action-item crm-hover-button\" title='Edit Recurring Payment' >Edit</a><a href=\"/index.php?q=civicrm/contribute/unsubscribe&amp;reset=1&amp;crid=" . $recurID . "&amp;cid=" . $contactID . "&amp;context=contribution\" class=\"action-item crm-hover-button\" title='Cancel' >Cancel</a></span>",
+      $this->getActionHtml()
+    );
+  }
+
+  /**
+   * Test links render correctly for manual processor.
+   *
+   * @throws \API_Exception
+   * @throws \CiviCRM_API3_Exception
+   */
+  public function testLinksPaypalStandard(): void {
+    $this->setupTemplate([
+      'payment_processor_id' => $this->paymentProcessorCreate(['payment_processor_type_id' => 'PayPal_Standard']),
+      'contact_id' => $this->createLoggedInUser(),
+    ]);
+    $expected = '<span><a href="/index.php?q=civicrm/contact/view/contributionrecur&amp;reset=1&amp;id=1&amp;cid=3&amp;context=contribution" class="action-item crm-hover-button" title=\'View Recurring Payment\' >View</a><a href="/index.php?q=civicrm/contribute/updaterecur&amp;reset=1&amp;action=update&amp;crid=1&amp;cid=3&amp;context=contribution" class="action-item crm-hover-button" title=\'Edit Recurring Payment\' >Edit</a></span><span class=\'btn-slide crm-hover-button\'>more<ul class=\'panel\'><li><a href="/index.php?q=civicrm/contribute/unsubscribe&amp;reset=1&amp;crid=1&amp;cid=3&amp;context=contribution" class="action-item crm-hover-button" title=\'Cancel\' >Cancel</a></li><li><a href="/index.php?q=civicrm/contribute/updatebilling&amp;reset=1&amp;crid=1&amp;cid=3&amp;context=contribution" class="action-item crm-hover-button" title=\'Change Billing Details\' >Change Billing Details</a></li></ul></span>';
+    $this->assertEquals($expected, $this->getActionHtml());
+
+    $page = new CRM_Contribute_Page_UserDashboard();
+    $page->run();
+    $expected = '<span><a href="https://www.sandbox.paypal.com/cgi-bin/webscr?cmd=_subscr-find&alias=sunil._1183377782_biz_api1.webaccess.co.in" class="action-item crm-hover-button" title=\'Cancel\' >Cancel</a>'
```

Comment:
@eileenmcnaughton test failure seems to be due to the missing class in this line. https://test.civicrm.org/job/CiviCRM-Core-PR/40862/testReport/junit/(root)/CRM_Contribute_Page_TabTest/testLinksPaypalStandard/